### DISCUSSION TITLE
Feedback: needs more documentation

### DISCUSSION THREAD

**Discussion by thany** [typescript-eslint/typescript-eslint#10413] [Community Feedback]
### Body

Just some feedback. The documentation needs to be clearer.

Please consider this eslint config:

```
// @ts-check
import turbo from 'eslint-config-turbo/flat';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  ...turbo,
  {
    // ...other irrelevant config
  }
);
```

It reports this error:

```
Argument of type '{ plugins: { turbo: typeof plugin; }; rules: { "turbo/no-undeclared-env-vars": string; }; }' is not assignable to parameter of type 'InfiniteDepthConfigWithExtends'.
  Types of property 'rules' are incompatible.
    Type '{ "turbo/no-undeclared-env-vars": string; }' is not assignable to type 'Partial<Record<string, RuleEntry>>'.
      Property '"turbo/no-undeclared-env-vars"' is incompatible with index signature.
        Type 'string' is not assignable to type 'RuleEntry | undefined'.ts(2345)
```

I have no way of knowing how to actually do this. The documentation doesn't go beyond the simplest possible scenario, and fails to consider basically anything else.

I can't just do `turbo as ConfigWithExtends` or whatever, because ESLint requires the config to be a javascript file. So in fact there's literally nothing I can add to make typescript understand what it's importing.

And also, I'm not the author of `eslint-config-turbo` so I can't just add typings to it. But moreover, you can't expect every author of every eslint config package to provide their typings.

Afaict, this is placing me between a rock and a hard place. Please update the documentation with more scenarios like this one.

### Additional Info

Sorry about my last topic. I didn't know I wasn't allowed to ask about a problem - I consider that feedback. Kinda rude to close it without allowing me to explain myself, so I just created a new discussion, more clearly formulating it as feedback. Please do not close and lock 🙏🏻

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
As clearly mentioned in our discussion guidelines

> Community Feedback: Questions from the maintainer team that should be raised to the broader community

As you are not in the maintainer team, nor are you asking a question of the broader community - using a discussion is incorrect.

Your feedback is valuable - but we need it via the correct channel.
If you want to raise a request for documentation - then raise an issue. There is an issue template for exactly this:
https://github.com/typescript-eslint/typescript-eslint/issues/new?assignees=&labels=documentation%2Ctriage&projects=&template=05-documentation-request.yml&title=Docs%3A+%3Ca+short+description+of+my+proposal%3E
